### ISSUE TITLE
Liveview ignores registered template engines when using implict render

### ISSUE THREAD

**Issue by tensiondriven** [phoenixframework/phoenix_live_view#1817]
I'm using LiveView with the `slime` template language.  I extended `slime` and `phoenix_slime` deps to output HEEx, which works swimmingly with an explicit render call, but when attempting to use implicit render (that is, a Liveview with no `render/1` implementation), Liveview ignores my `.sheex` template, giving this error:

```
    ** (RuntimeError) render/1 was not implemented for InboxheroWeb.TableLive.

Make sure to either explicitly define a render/1 clause with a LiveView template:

    def render(assigns) do
      ~H"""
      ...
      """
    end

Or create a file at "/Users/j/Code/[app name]/lib/[app name]_web/liveviews/table_live.html.heex" with the LiveView template.
```

I tried using filenames with the following extensions:

`.html.sheex`
`.html.heex.sheex`

Note:  `.sheex` is registered as a template engine and works when calling `render/2` via view:

```
config :phoenix, :template_engines,
  sheex: PhoenixSlime.LiveViewEngine
```

To reproduce this fully, you'll need to register a template engine that can compile down to HEEx.

This fork of `phoenix_slime` at https://github.com/tensiondriven/phoenix_slime adds support for HEEx to `phoenix_slime`, along with a counterpart fork of `slime` which is here: https://github.com/tensiondriven/slime.  There's an open PR against the official `slime` hex dep using this branch which can be seen here: https://github.com/slime-lang/slime/pull/168

### Environment

* Elixir version (elixir -v): `Elixir 1.12.2 (compiled with Erlang/OTP 22)`
* Phoenix version (mix deps): `phoenix 1.6.5`
* Phoenix LiveView version (mix deps): `phoenix_live_view 0.17.5`
* Operating system: macOS Monterey 12.1 (21C52)
* Browsers you attempted to reproduce this bug on (the more the merrier):  Chrome (N/A)
* Does the problem persist after removing "assets/node_modules" and trying again? Yes/no: Yes (N/A)

### Actual behavior

Phoenix throws an error, ignoring the registered template engine:

```
    ** (RuntimeError) render/1 was not implemented for InboxheroWeb.TableLive.

Make sure to either explicitly define a render/1 clause with a LiveView template:

    def render(assigns) do
      ~H"""
      ...
      """
    end

Or create a file at "/Users/j/Code/inboxhero/lib/inboxhero_web/liveviews/table_live.html.heex" with the LiveView template.
```

### Expected behavior

Phoenix should respect registered template engines when attempting implicit render.

**Comment by josevalim** [comment]
The code is meant to be template engine agnostic: https://github.com/phoenixframework/phoenix_live_view/blob/master/lib/phoenix_live_view/renderer.ex#L10 - so it should all work.

We only use `.heex` for showing the error message: https://github.com/phoenixframework/phoenix_live_view/blob/master/lib/phoenix_live_view/renderer.ex#L45

Try calling `Phoenix.Template.engines()` in IEx and see what it returns.

**Comment by tensiondriven** [comment]
Thanks @josevalim – I think I got to the bottom of it after significant digging.  Liveview is indeed detecting templates for all registered engines, but only at compile time, and the behavior is different for `.html.heex` vs `.html.sheex` (in my example).

What threw me is that changes to the template's's name, or its existence, are not picked up upon refresh, or even server restarts, unless I touch the liveview's .ex file.  Presumably this is because the templates are being detected at compile time only.  This is particularly confusing as a developer, because the following situation can happen easily:

1) Create a liveview with implicit render and no template
2) Run the server
An error is displayed (as expected)
4) Create the template using an extension from a rendering engine, refresh the page
* The error about missing template persists
5) Stop the server, restart it, and refresh the page
* The error about missing template persists

Touching the liveview file, causing it to recompile, will cause a re-check for the template, and then the template will be used, but this isn't obvious.

What makes it more confusing is that if one creates a `.html.heex` file, the new file is picked up immedaitely.  Only when using a file from a different template engine is the new file ignored.

Could the error message be amended to include something about the need to recompile the liveview, or could this be fixed by checking for templates again at runtime before raising?

Here's my engines list, fwiw:

```
➜  inboxhero git:(master) ✗ iex -S mix
Erlang/OTP 24 [erts-12.0.2] [source] [64-bit] [smp:4:4] [ds:4:4:10] [async-threads:1] [jit]

[debug] Loading Initial Jobs from Config
Interactive Elixir (1.12.2) - press Ctrl+C to exit (type h() ENTER for help)
iex(1)> Phoenix.Template.engines()
%{
  eex: Phoenix.Template.EExEngine,
  exs: Phoenix.Template.ExsEngine,
  heex: Phoenix.LiveView.HTMLEngine,
  leex: Phoenix.LiveView.Engine,
  sheex: PhoenixSlime.LiveViewHTMLEngine,
  slim: PhoenixSlime.Engine,
  slime: PhoenixSlime.Engine
}
```

**Comment by josevalim** [comment]
The issue about files not being picked up has been fixed on Elixir v1.13 :)

**Comment by josevalim** [comment]
I think this can be closed then, thanks for the follow up!
